Assets/_BForBoss/_Core/Scripts/HUD/CountdownViewBehaviour.cs

```diff
@@ -0,0 +1,83 @@
+using System;
+using Perigon.Utility;
+using TMPro;
+using UnityEngine;
+
+namespace BForBoss
+{
+    public class CountdownViewBehaviour : MonoBehaviour
+    {
+        [SerializeField, Resolve] private TMP_Text _countdownLabel;
+        private float _time;
+        private bool _isRunning;
+
+        private float _amountOfTime;
+        private Action _onCountdownComplete;
+
+        public void InitializeCountdown(float amountOfTime, Action onCountdownComplete)
```

Comment:
You can probably remove this method. 

And move the `amountOfTime` into `StartCountdown` instead. 
So whoever starts it can change it and start it that amount of time.